evolver/hardware/standard/od_sensor.py

```diff
@@ -0,0 +1,35 @@
+from pydantic import Field
+
+from evolver.connection.interface import Connection
+from evolver.hardware.interface import SensorDriver, VialBaseModel, VialConfigBaseModel
+from evolver.serial import SerialData
+
+
+class ODSensor(SensorDriver):
+    """Optical density sensor driver
+
+    This driver represents a family of sensors for turbidity which are read by
+    integrating a specified number of ADC readings per vial.
+    """
+
+    class Config(VialConfigBaseModel):
+        addr: str = Field(description="Address of od sensor on serial bus (e.g. od_90)")
+        integrations: int = Field(500, description="on read, request average of this number of ADC reads")
+        serial_conn: Connection | None = Field(None, description="serial connection, default is that on evolver")
+
+    class Output(VialBaseModel):
+        raw: int
+        density: float = None
+
+    @property
+    def serial(self):
+        return self.serial_conn or self.evolver.serial
+
+    def read(self):
+        self.outputs = {}
+        cmd = SerialData(addr=self.addr, data=[str(self.integrations).encode()], kind="r")
+        with self.serial as comm:
+            response = comm.communicate(cmd)
+        for vial, raw in enumerate(response.data):
```

Comment:
Remind me, is it guaranteed that this is ordered by vial such that we can just enumerate here?

Reply:
yes, it will come back in order - in fact the position in the response is the only indicator of vial number